File: avancado-intermediario/Python/Q4/universidade_ead.py
from universidade import Universidade

class UniversidadeEaD(Universidade):

    def __init__(self, nome, urlPlataforma):
        super().__init__(nome)
        self.__urlPlataforma = urlPlataforma

    def getUrlPlataforma(self):
        return self.__urlPlataforma

    def setUrlPlataforma(self, urlPlataforma):
        self.__urlPlataforma = urlPlataforma

    def aplicarAvaliacao(self):
        print(f"Avaliações Online na {self.getUrlPlataforma()}")

    def __str__(self) -> str:
        return f"UniversidadeEaD: {self.getNome()} | URL: {self.getUrlPlataforma()}"

File: avancado-intermediario/Java/Q4/UniversidadeEaD.java
public class UniversidadeEaD extends Universidade{
    private String urlPlataforma;

    public UniversidadeEaD(String nome, String urlPlataforma){
        super(nome);
        this.urlPlataforma = urlPlataforma;
    }

    public String getUrlPlataforma(){
        return this.urlPlataforma;
    }

    public void setUrlPlataforma(String urlPlataforma){
        this.urlPlataforma = urlPlataforma;
    }

    public void aplicarAvaliacao(){
        System.out.println("Avaliações Online na " + getUrlPlataforma());
    }

    @Override
    public String toString() {
        return super.toString() + " (EaD) | Plataforma: " + urlPlataforma;
    }
}